mmv1/third_party/terraform/tpgresource/common_diff_suppress.go.erb

```diff
@@ -281,6 +281,23 @@ func LastSlashDiffSuppress(_, old, new string, _ *schema.ResourceData) bool {
 	return new == old
 }
 
+// Suppress diffs when the value read from api
+// has the project number instead of the project name.
+// This function suppresses the difference when the project ID is not prefixed with the string "projects".
+func ProjectNumberDiffSuppressWithoutPrefix(k, old, new string, _ *schema.ResourceData) bool {
```

Comment:
Could you implement this in services/alloydb (using a [`custom_code.constants`](https://googlecloudplatform.github.io/magic-modules/develop/custom-code/#add-reusable-variables-and-functions) file) instead of in common_diff_suppress? Although it is similar to ProjectNumberDiffSuppress it may not be as widely applicable to different services, so I'd rather not add it globally.

It would also be great to get some [unit tests](https://googlecloudplatform.github.io/magic-modules/develop/test/test/#add-unit-tests) for this function.